lib/gridfs-stream/upload.js

```diff
@@ -92,6 +94,36 @@ GridFSBucketWriteStream.prototype.write = function(chunk, encoding, callback) {
 };
 
 /**
+ * Places this write stream into an aborted state (all future writes fail)
+ * and deletes all chunks that have already been written.
+ *
+ * @method
+ * @param {GridFSBucket~errorCallback} callback called when chunks are successfully removed or error occurred
+ * @return {Promise} if no callback specified
+ */
+
+GridFSBucketWriteStream.prototype.abort = function(callback) {
+  if (this.state.streamEnd) {
+    var error = new Error('Cannot abort a stream that has already completed');
+    if (callback) {
+      return callback(error);
+    }
+    return this.state.promiseLibrary.reject(error);
+  }
+  if (this.state.aborted) {
+    var error = new Error('Cannot call abort() on a stream twice');
+    if (callback) {
+      return callback(error);
+    }
+    return this.state.promiseLibrary.reject(error);
+  }
+  this.state.aborted = true;
+  this.chunks.deleteMany({ files_id: this.id }, function(error) {
+    callback(error);
```

Comment:
`callback` is not tested for existence here. The rest of the function does test this.
If this function accepts an optional callback, then it should be documented as optional too, else the other checks are not needed.